### ISSUE TITLE
12099 Flash Hesitation When Editing Tech

### ISSUE THREAD

**Issue by beelee1** [triplea-game/triplea#4062]
### My Operating System
Linux

### Map name
Global 40 House Rules with Canada

### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?)
 When adding tech via edit it does a 1 sec or so flash. Similar to the previous flash things when scrolling I think it was. Can't remember for sure. Might have been something else

### Instead of this error, what should have happened?
Shouldn't flash.

### Any additional information that may help

Not a huge deal as most maps don't use tech via edit. This one does or can though. Don't remember it happening before, so that's why I mention it. I'll attach a save
[sep 19 2018 Flash on Tech Add.zip](https://github.com/triplea-game/triplea/files/2399344/sep.19.2018.Flash.on.Tech.Add.zip)

**Comment by beelee1** [comment]
yea doesn't do it in 11440. 
SBR froze the game up in 12099 too. I've time to properly diagnose it as Dan Van suggested a while back with a similar issue. I'll get back with the error for that

**Comment by ssoloff** [comment]
@beelee1 I just want to make sure I understand what you mean by "flash hesitation."  When I add a tech on that map, I see the map get cleared and redrawn.  Is that what you're seeing?

**Comment by beelee1** [comment]
yea that's what's happening. Real brief, maybe not even a second, but as I said above doesn't do it at all in 11440

**Comment by ssoloff** [comment]
@beelee1 Thanks for confirming!

I actually do see it in 11440.  I think it might be because I'm running on a VM, and that tends to exacerbate any timing issues so that I notice things on it that I don't normally see on my physical box.

I went back through several releases and think I found the point where this issue was introduced.  10498 was the last build that doesn't flicker for me, while 10506 is the first build that does flicker.  That implies that #3514 would have introduced the flicker, and looking at those changes, it kind of makes sense.

@ron-murhammer @RoiEXLab If I'm reading that code right, it means anytime the game data changes, the map will be redrawn, correct?  Is the flicker reported in this PR an acceptable side effect for the correctness presumably #3514 was meant to ensure?  If not, is there a way we can double-buffer the redraw operation to avoid clearing the map before drawing the tiles?

**Comment by beelee1** [comment]
Right on  That VM must be badass : ) I can't get it to do it on 11440. 
Rock on : )

**Comment by ron-murhammer** [comment]
@ssoloff Yeah, this is known. It isn't ideal but never got back around to trying to figure out how to do it better.

**Comment by beelee1** [comment]
just checked this on 2.5.22294 and it still happens. Pretty low priority but I'll leave it open since it's not fixed.
